tests/v2/integration/catalogv2/system_charts_test.go

```diff
@@ -0,0 +1,439 @@
+	t := 360
+	var app *rv1.App
+
+	err := kwait.Poll(PollInterval, time.Duration(t)*time.Second, func() (done bool, err error) {
+		app, err = w.catalogClient.Apps("cattle-system").Get(context.TODO(), name, metav1.GetOptions{})
+		e, ok := err.(*errors.StatusError)
+		if ok && errors.IsNotFound(e) {
+			return false, nil
+		}
+		if err != nil {
+			return false, err
+		}
+
+		if app.Spec.Info.Status == status && app.Spec.Version > previousVersion {
+			return true, nil
+		}
+		return false, nil
+	})
+	return app, err
+}
+
+func (w *SystemChartsTest) updateManagementCluster() error {
+	w.cluster.AKSConfig = &client.AKSClusterConfigSpec{}
+	_, err := w.client.Management.Cluster.Replace(w.cluster)
+	return err
+}
+
+func (w *SystemChartsTest) resetManagementCluster() error {
+	w.cluster.AKSConfig = nil
+	w.cluster.AppliedSpec.AKSConfig = nil
+	_, err := w.client.Management.Cluster.Replace(w.cluster)
+	err = kwait.Poll(2*time.Second, time.Minute, func() (done bool, err error) {
+		c, err := w.client.Management.Cluster.ByID("local")
+		if err != nil {
+			return false, err
+		}
+		if c.AKSConfig == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	return err
+}
+
+func (w *SystemChartsTest) updateSetting(name, value string) error {
+	// Use the Steve client instead of the main one to be able to set a setting's value to an empty string.
+	existing, err := w.client.Steve.SteveType("management.cattle.io.setting").ByID(name)
+	if err != nil {
+		return err
+	}
+
+	var s v3.Setting
+	if err := stevev1.ConvertToK8sType(existing.JSONResp, &s); err != nil {
+		return err
+	}
+
+	s.Value = value
+	_, err = w.client.Steve.SteveType("management.cattle.io.setting").Update(existing, s)
+	return err
+}
+
+func (w *SystemChartsTest) uninstallApp(namespace, chartName string) error {
+	var cfg action.Configuration
+	if err := cfg.Init(w.restClientGetter, namespace, "", logrus.Infof); err != nil {
+		return err
+	}
+	l := action.NewList(&cfg)
+	releases, err := l.Run()
+	l.StateMask = action.ListAll
+	l.SetStateMask()
+	if err != nil {
+		return fmt.Errorf("failed to fetch all releases in the %s namespace: %w", namespace, err)
+	}
+	for _, r := range releases {
+		if r.Chart.Name() == chartName {
+			err = kwait.Poll(10*time.Second, time.Minute, func() (done bool, err error) {
+				if _, err := action.NewUninstall(&cfg).Run(r.Name); err != nil {
+					return false, nil
+				}
+				return true, nil
+			})
+			return err
+		}
+	}
+	return nil
+}
+
+// pollUntilDownloaded Polls until the ClusterRepo of the given name has been downloaded (by comparing prevDownloadTime against the current DownloadTime)
+func (w *SystemChartsTest) pollUntilDownloaded(ClusterRepoName string, prevDownloadTime metav1.Time) error {
+	err := kwait.Poll(PollInterval, time.Minute, func() (done bool, err error) {
+		clusterRepo, err := w.client.Steve.SteveType(catalog.ClusterRepoSteveResourceType).ByID(ClusterRepoName)
```

Comment:
Can we do something like this here so that we avoid converting status to kubernetes type later at line no 429 ? 

clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})